### PR TITLE
Change signature to resolve error

### PR DESCRIPTION
**Pull request by mccalluc**
With this change, I no longer get the signature error in https://github.com/phovea/phovea_d3/issues/10,
and tests here still pass, but...
- Why wasn't the build failing here before?
- Apart from silencing errors, what is actually "right"?

### CODE REVIEW

**Comment by mccalluc** [comment]
```
src/provenance.ts(1122,12): error TS2322: Type 'Promise<GraphBase>' is not assignable to type 'Promise<ProvenanceGraph>'.
  Type 'GraphBase' is not assignable to type 'ProvenanceGraph'.
    Property '_actions' is missing in type 'GraphBase'.
```

Travis did run successfully before this change: https://travis-ci.org/phovea/phovea_core/builds/178331267

File: src/data.ts
Comment (thinkh) [inline src/data.ts:164]: @mccalluc This defines the result type of the promise. I don't know why the auto inference doesn't work in this case.

See also the `then()` definition:
```
    /**
     * Attaches callbacks for the resolution and/or rejection of the Promise.
     * @param onfulfilled The callback to execute when the Promise is resolved.
     * @returns A Promise for the completion of which ever callback is executed.
     */
    then<TResult>(onfulfilled: (value: T) => TResult | PromiseLike<TResult>): Promise<TResult>;
```

**Review by thinkh — APPROVED** [review]

File: src/data.ts
Comment (mccalluc) [reply src/data.ts:164]: Thanks. The other problem I quoted above still remains... I'll see if I can understand it better this morning.

**Comment by thinkh** [comment]
@mccalluc I fixed all typescript typing errors. You can merge this PR and the related PRs https://github.com/phovea/phovea_d3/issues/11 and https://github.com/phovea/phovea_vis/issues/14. Please close the related issues, too. Thanks.

**Comment by mccalluc** [comment]
Travis is still failing. This should not be merged.

**Comment by thinkh** [comment]
@mccalluc Yes, Travis still fails, but of another error: `TypeError: Cannot read property 'exclude' of undefined`

**Comment by mccalluc** [comment]
I think we want to make sure everything passes before we merge: At some point the build process worked, presumably, and its current state seems to be a regression.... That said, I actually can't reproduce the failure locally.
```
webpack: wait until bundle finished: 
Hash: 9f9118a59fe90787392d
Version: webpack 2.1.0-beta.27
Time: 3748ms
```
All the more reason to sort this out now. We should also increment the version number. And, do we have any idea why this broke now? I'll see if I can understand the build error.

**Comment by thinkh** [comment]
@mccalluc I can reproduce the Travis errors locally, but I don't have a solution.

@sgratzl Do know what could be the problem for the testing/[Travis errors](https://travis-ci.org/phovea/phovea_core/builds/183609720)?

**Comment by sgratzl** [comment]
see https://github.com/Caleydo/lineup.js/commit/826715ed7060f707edf273dfb3e235df479ab4ac

**Comment by mccalluc** [comment]
Thanks! Suggestion: Let's put this on hold, and I'll make a new PR which only pins the TS version number, and we'll see how that works. Ideally, if I pin it to a sufficiently old version that will fix the problems we're seeing.

**Comment by thinkh** [comment]
@sgratzl Thanks for the hint.

@mccalluc Ok, you can do that. I updated my local Typescript to v2.1.4 and I'm still getting the error.

**Comment by mccalluc** [comment]
With the merge from master, Travis is happy. I don't think the other changes get in the way: They'll be needed if/when we do upgrade. This PR was already reviewed and approved, so I'll merge.

**Pull request merged**
